rcl/src/rcl/logging_rosout.c

```diff
@@ -146,6 +247,10 @@ rcl_ret_t rcl_logging_rosout_init_publisher_for_node(rcl_node_t * node)
   RCL_CHECK_ARGUMENT_FOR_NULL(node, RCL_RET_NODE_INVALID);
   logger_name = rcl_node_get_logger_name(node);
   if (NULL == logger_name) {
+    // Prevent from printing an extra error information in some tests
```

Comment:
Can you please comment on what tests were printing this extra error information?

Reply:
When I run the test with `./test_logging_rosout__rmw_fastrtps_cpp --gtest_filter="*.test_bad_params_init_fini_node_publisher"`, the test outputs as follows,

```
chenlh test $ ./test_logging_rosout__rmw_fastrtps_cpp --gtest_filter="*.test_bad_params_init_fini_node_publisher"
Running main() from /home/chenlh/Projects/ROS2/ros2-master/local_install_test/gtest_vendor/src/gtest_vendor/src/gtest_main.cc
Note: Google Test filter = *.test_bad_params_init_fini_node_publisher
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from TestLogRosoutFixtureNotParam__rmw_fastrtps_cpp
[ RUN      ] TestLogRosoutFixtureNotParam__rmw_fastrtps_cpp.test_bad_params_init_fini_node_publisher

>>> [rcutils|error_handling.c:108] rcutils_set_error_state()
This error state is being overwritten:

  'rcl node implementation is invalid, at /home/chenlh/Projects/ROS2/ros2-master/src/ros2/rcl/rcl/src/rcl/node.c:414'

with this new error message:

  'Logger name was null., at /home/chenlh/Projects/ROS2/ros2-master/src/ros2/rcl/rcl/src/rcl/logging_rosout.c:249'

rcutils_reset_error() should be called after error handling to avoid this.
<<<
[       OK ] TestLogRosoutFixtureNotParam__rmw_fastrtps_cpp.test_bad_params_init_fini_node_publisher (0 ms)
[----------] 1 test from TestLogRosoutFixtureNotParam__rmw_fastrtps_cpp (0 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (0 ms total)
[  PASSED  ] 1 test.
```
I expected a clean output with only gtest information.

Considering that it's not what this PR should do, I'll remove this fix, because I think this issue exists most places.